expression/typeinferer.go

```diff
@@ -367,6 +367,18 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
 			tp.Charset, tp.Collate = types.DefaultCharsetForType(tp.Tp)
 		}
 	// number related
+	case ast.AddTime:
+		t := x.Args[0].GetType().Tp
+		switch t {
+		case mysql.TypeDatetime,mysql.TypeDate,mysql.TypeTimestamp:
```

Comment:
`args[0].Tp == mysql.TypeDate` should be contained in `default`
and the `chs` should be setted to 'utf8' in the `default` case.